gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -98,6 +112,17 @@ public synchronized int read() throws IOException {
     return (b & 0xff);
   }
 
+  private void captureDuration(String suffix, Stopwatch stopwatch, int byteRead) {
+    if (isTraceLoggingEnabled) {
+      Map<String, Object> jsonMap = new HashMap<>();
+      // Capturing time in nanos because majority of the reads will be done from cache
+      long nanos = stopwatch.elapsed(TimeUnit.NANOSECONDS);
+      jsonMap.put(suffix + "_latencyNs", nanos);
+      jsonMap.put(suffix + "_byteRead", byteRead);
```

Comment:
please have "suffix" as a separate field and name it "method" so that it is easier to query.

Reply:
Done